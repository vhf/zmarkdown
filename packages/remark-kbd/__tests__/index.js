import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import remarkStringify from 'remark-stringify'
import rehypeStringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'
import remarkCustomBlocks from '../../remark-custom-blocks'

import plugin from '../src/'


const render = text => unified()
  .use(reParse, {
    footnotes: true
  })
  .use(remark2rehype)
  .use(remarkCustomBlocks, {
    secret: 'spoiler'
  })
  .use(plugin)
  .use(rehypeStringify)
  .processSync(text)

const fixture = dedent`
  Blabla ||ok|| kxcvj ||ok foo|| sdff

  sdf |||| df

  sfdgs | | dfg || dgsg | qs

  With two pipes: \||key|| you'll get ||key||.

  It parses inline elements inside:

  * ||hell[~~o~~](#he)?||

  but not block elements inside:

  * ||hello: [[secret]]?||
`


test('kbd', () => {
  const { contents } = render(fixture)
  expect(contents).toMatchSnapshot()
})

test('to markdown', () => {
  const { contents } = unified()
    .use(reParse)
    .use(remarkStringify)
    .use(plugin)
    .processSync(fixture)

  expect(contents).toMatchSnapshot()
})
