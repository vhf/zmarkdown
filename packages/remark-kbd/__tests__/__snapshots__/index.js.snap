// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kbd 1`] = `
"<p>Blabla <kbd>ok</kbd> kxcvj <kbd>ok foo</kbd> sdff</p>
<p>sdf |||| df</p>
<p>sfdgs | | dfg || dgsg | qs</p>
<p>With two pipes: ||key|| you'll get <kbd>key</kbd>.</p>
<p>It parses inline elements inside:</p>
<ul>
<li><kbd>hell<a href=\\"#he\\"><del>o</del></a>?</kbd></li>
</ul>
<p>but not block elements inside:</p>
<ul>
<li><kbd>hello: [[secret]]?</kbd></li>
</ul>"
`;

exports[`to markdown 1`] = `
"Blabla ||ok|| kxcvj ||ok foo|| sdff

sdf |||| df

sfdgs | | dfg || dgsg | qs

With two pipes: \\\\||key|| you'll get ||key||.

It parses inline elements inside:

-   ||hell[~~o~~](#he)?||

but not block elements inside:

-   ||hello: \\\\[[secret]]?||
"
`;
